ziti-springboot-client/src/main/java/org/openziti/springboot/client/web/httpclient/ZitiConnectionSocketFactory.java

```diff
@@ -82,18 +82,14 @@ public ZitiConnectionSocketFactory(ZitiContext ctx) {
 
   @Override
   public Socket createSocket(HttpContext context) throws IOException {
-    if (socketFactory == null) {
-      this.socketFactory = Ziti.getSocketFactory();
-    }
-    return socketFactory.createSocket();
+    return Ziti.getSocketFactory().createSocket();
   }
 
   @Override
   public Socket connectSocket(TimeValue timeValue, Socket socket, HttpHost host, InetSocketAddress inetSocketAddress, InetSocketAddress localAddress, HttpContext context) throws IOException {
 
     final Socket sock = socket != null ? socket : createSocket(context);
-    // can leave InetSocketAddress as unresolved since ziti performs a service lookup using the host and port
-    sock.connect(InetSocketAddress.createUnresolved(host.getHostName(), host.getPort()), timeValue.toMillisecondsIntBound());
+    sock.connect(inetSocketAddress, timeValue.toMillisecondsIntBound());
```

Comment:
will this work properly for unresolvable addresses? `createUnresolved` seems to be important since "No attempt will be made to resolve the hostname into an InetAddress"

Did you confirm this won't throw any error with ficticious/unresolvable addresses?

Reply:
It returns null if it can't resolve it and it falls back to the os resolver below.